### PR TITLE
Allows disabling token propagation override.

### LINKED ISSUE #84: Token propagation should't override existing token in clients
Authorization header is getting unexpectedly overridden by the token propagation.

### PR DESCRIPTION
**Pull request by alvarosanchez**
Fixes #84

### CODE REVIEW

File: security/src/main/java/io/micronaut/security/token/propagation/TokenPropagationHttpClientFilter.java [hunk 1]
```diff
@@ -92,11 +112,25 @@ public TokenPropagationHttpClientFilter(TokenWriter tokenWriter,
...
         return chain.proceed(targetRequest);
     }
+
+    private boolean hasExistingToken(MutableHttpRequest<?> targetRequest) {
+        String headerName = HttpHeaders.AUTHORIZATION;
+        if (httpHeaderTokenWriterConfiguration != null) {
+            headerName = httpHeaderTokenWriterConfiguration.getHeaderName();
+        }
+        return targetRequest.getHeaders().get(headerName) != null;
+    }
```
Comment (jameskleeh) [inline security/src/main/java/io/micronaut/security/token/propagation/TokenPropagationHttpClientFilter.java]: This assumes the token will be written via headers. It should be the responsibility of the token writer to determine if the token already exists.

**Comment by jameskleeh** [comment]
Also this is a new feature so this should target 1.3

File: security/src/main/java/io/micronaut/security/token/propagation/TokenPropagationHttpClientFilter.java (hunk 1 above)
Comment (alvarosanchez) [reply security/src/main/java/io/micronaut/security/token/propagation/TokenPropagationHttpClientFilter.java]: Good point, but actually now you mention, it's `TokenReader` the one that should determine whether the token exists

**Comment by alvarosanchez** [comment]
Targeted master and used `TokenReader` to find the token.

The build will fail because master is broken.

**Comment by jameskleeh** [comment]
Manually applied these changes to 1.4.x without a setting

**Pull request closed without merging**